nobo.py

```diff
@@ -452,3 +470,69 @@ def response_handler(self, r):
+        command = [self.API.ADD_OVERRIDE, '1', mode, type, end_time, start_time, target_type, target_id]
+        self.send_command(command)
+    
+    # Function to update name, week profile, temperature or override allowing for a zone
+    def update_zone(self, zone_id, name=None, week_profile_id=None, temp_comfort_c=None, temp_eco_c=None, override_allowed=None):
+        # Initialize command with the current zone settings
+        command = [self.API.UPDATE_ZONE] + list(self.zones[zone_id].values()) + [self.API.OVERRIDE_ID_NONE]
+        
+        # Replace command with arguments that are not None. Is there a more elegant way?
+        if name:
+            command[2] = name
+        if week_profile_id:
+            command[3] = week_profile_id
+        if temp_comfort_c:
+            command[4] = temp_comfort_c
+        if temp_eco_c:
+            command[5] = temp_eco_c
+        if override_allowed:
+            command[6] = override_allowed
+
+        self.send_command(command)
+
+    # Function to find the status of a profile at a certain time in the week. Monday is day 0
+    def get_week_profile_status(self, profile, dt):
+        target = (dt.hour*100) + dt.minute
+        # profile[0] is always 0000x, so this provides the initial status
+        status = profile[0][-1]
+        weekday = 0
+        for timestamp in profile[1:]:
+            if timestamp[:4] == '0000':
+                weekday += 1
+            if weekday == dt.weekday():
+                if int(timestamp[:4]) <= target:
+                    status = timestamp[-1]
+                else:
+                    break
+        self.logger.debug('Status at {} on weekday {} is {}'.format(dt.weekday(), target, status))
+        return status
+
+	# Function to find mode in a zone right now
+    def get_current_zone_mode(self, zone_id, now=datetime.datetime.today()):
+        current_time = (now.hour*100) + now.minute
+        current_mode = ''
+        
+        if self.zones[zone_id]['override_allowed'] == '1':
+            for o in self.overrides:
+                if self.overrides[o]['mode'] == '0':
+                    continue # "normal" overrides
+                elif self.overrides[o]['target_type'] == self.API.OVERRIDE_TARGET_ZONE:
+                    if self.overrides[o]['target_id'] == zone_id:
+                        current_mode = self.API.DICT_OVERRIDE_MODE_TO_NAME[self.overrides[o]['mode']]
+                        break
+                elif self.overrides[o]['target_type'] == self.API.OVERRIDE_TARGET_GLOBAL:
+                    current_mode = self.API.DICT_OVERRIDE_MODE_TO_NAME[self.overrides[o]['mode']]
+
+        # no override - figure out from week profile
+        if not current_mode:
+            current_profile = self.week_profiles[self.zones[zone_id]['week_profile_id']]['profile']
+            current_status = self.get_week_profile_status(current_profile, now)
+            current_mode = self.API.DICT_WEEK_PROFILE_STATUS_TO_NAME[current_status]
```

Comment:
Should this be done inside the get_week_profile_status,?

I assume that function can be useful to call for others, and it would be nice  to always use names exterally, the numbers differ between week profiles and overrides and can be confusing.